code/modules/projectiles/guns/projectile/siren.dm

```diff
@@ -75,14 +80,79 @@
 			in_chamber.reagents.trans_to(src, in_chamber.reagents.total_volume)
 		qdel(in_chamber)
 		in_chamber = null
-	in_chamber = new/obj/item/projectile/bullet/liquid_blob(src, hard)
+	in_chamber = new projectile_type(src, hard)
 	reagents.trans_to(in_chamber, 10)
 	if(!hard) //When set to no-damage mode, each shot has five times the reagents.
 		for(var/datum/reagent/R in in_chamber.reagents.reagent_list)
 			in_chamber.reagents.add_reagent(R.id, reagents.get_reagent_amount(R.id)*4)
 	Fire(A,user,params, "struggle" = struggle)
 
 /obj/item/weapon/gun/siren/process_chambered()
+	return in_chamber
+
+/obj/item/weapon/gun/siren/supersoaker
+	name = "super soaker"
+	desc = "For ages 10 and up."
+	icon_state = "super_soaker"
+	item_state = "gun"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/items_lefthand.dmi', "right_hand" = 'icons/mob/in-hand/right/items_righthand.dmi')
+	origin_tech = Tc_COMBAT + "=1"
+	recoil = 0
+	flags = FPRINT | OPENCONTAINER
+	fire_sound = 'sound/items/egg_squash.ogg'
+	max_reagents = 200
+	hard = -1
+	projectile_type = /obj/item/projectile/beam/liquid_stream
+	clumsy_check = 0
+	var/pumping = FALSE
+	var/pumps = 0
+
+/obj/item/weapon/gun/siren/supersoaker/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	if(flag)
+		return //we're placing gun on a table or in backpack
+	if(harm_labeled >= min_harm_label)
+		to_chat(user, "<span class='warning'>A label sticks the trigger to the trigger guard!</span>")//Such a new feature, the player might not know what's wrong if it doesn't tell them.
+		return
+	if(reagents.total_volume < 10 && !in_chamber)
+		return click_empty(user)
+	if(!in_chamber)
+		in_chamber = new projectile_type(src, max(3+(round(pumps/2)),15))
+		reagents.trans_to(in_chamber, 10)
+	Fire(A,user,params, struggle = struggle)
+	if(reagents.total_volume >= 10)
+		in_chamber = new projectile_type(src)
+		reagents.trans_to(in_chamber, 10)
+
+/obj/item/weapon/gun/siren/supersoaker/attack_self(mob/user as mob)
+	if(!pumping)
```

Comment:
Is this supposed to act as a sort of delayer? There's better ways to handle this. Use world time timestamps.